uber/models.py

```diff
@@ -1090,6 +1091,15 @@ def _badge_adjustments(self):
                     self.badge_num = self.session.next_badge_num(self.badge_type, old_badge_num=0)
 
     @presave_adjustment
+    def _status_adjustments(self):
+        if self.badge_status == c.NEW_STATUS and not self.placeholder and self.first_name:
+            if (self.paid == c.HAS_PAID or self.paid == c.NEED_NOT_PAY) or (self.paid == c.PAID_BY_GROUP and not self.group.amount_unpaid):
+                self.badge_status = c.COMPLETED_STATUS
+        elif self.badge_status == c.INVALID_STATUS and self.admin_account:
```

Comment:
Is this ``Tracking`` call necessary here?  Shouldn't that happen automatically when the thing is deleted?

Or does that not happen automatically because we're already inside the on-update event?